torchgeo/trainers/change.py

```diff
@@ -69,40 +70,92 @@ def __init__(
                 None for random weights, or the path to a saved model state dict. FCN
                 model does not support pretrained weights.
             in_channels: Number of channels per image.
+            task: One of 'binary', 'multiclass', or 'multilabel'.
+            num_classes: Number of prediction classes (only for ``task='multiclass'``).
+            num_labels: Number of prediction labels (only for ``task='multilabel'``).
+            num_filters: Number of filters. Only applicable when model='fcn'.
             pos_weight: A weight of positive examples and used with 'bce' loss.
             loss: Name of the loss function, currently supports
-                'bce', 'jaccard', or 'focal' loss.
+                'ce', 'bce', 'jaccard', and 'focal' loss.
+            class_weights: Optional rescaling weight given to each
+                class and used with 'ce' loss.
+            ignore_index: Optional integer class index to ignore in the loss and
+                metrics.
             lr: Learning rate for optimizer.
             patience: Patience for learning rate scheduler.
             freeze_backbone: Freeze the backbone network to fine-tune the
                 decoder and segmentation head.
             freeze_decoder: Freeze the decoder network to linear probe
                 the segmentation head.
-            num_filters: Number of filters. Only applicable when model='fcn'.
         """
         self.weights = weights
         super().__init__()
 
     def configure_losses(self) -> None:
         """Initialize the loss criterion."""
+        ignore_index: int | None = self.hparams['ignore_index']
+        class_weights = self.hparams['class_weights']
+        if class_weights is not None and not isinstance(class_weights, Tensor):
+            class_weights = torch.tensor(class_weights, dtype=torch.float32)
+
         match self.hparams['loss']:
+            case 'ce':
+                ignore_value = -1000 if ignore_index is None else ignore_index
+                self.criterion: nn.Module = nn.CrossEntropyLoss(
+                    ignore_index=ignore_value, weight=class_weights
+                )
             case 'bce':
                 self.criterion = nn.BCEWithLogitsLoss(
                     pos_weight=self.hparams['pos_weight']
                 )
             case 'jaccard':
-                self.criterion = smp.losses.JaccardLoss(mode='binary')
+                # JaccardLoss requires a list of classes to use instead of a class
+                # index to ignore.
+                if self.hparams['task'] == 'multiclass' and ignore_index is not None:
```

Comment:
Should ignore_index also apply to multilabel?

Reply:
I'm not convinced SemanticSegmentationTask is correct either, let's ignore this for now and fix it another day